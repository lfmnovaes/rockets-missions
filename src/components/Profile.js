import React from 'react';
import { Container, ListGroup } from 'react-bootstrap';
import { useSelector } from 'react-redux';

export default function Profile() {
  const missionsStore = useSelector((store) => store.missionsReducer);
  const reservedMissions = missionsStore.filter((mission) => mission.reserved === true);

  const rocketsStore = useSelector((store) => store.rocketsReducer);
  const reservedRockets = rocketsStore.filter((rocket) => rocket.reserved === true);

  return (
    <Container className="d-flex justify-content-evenly">
      <ListGroup>
        <h2>My Rockets</h2>
        {reservedRockets.map((rocket) => (
          <ListGroup.Item key={rocket.id}>{rocket.rocket_name}</ListGroup.Item>
        ))}
      </ListGroup>
      <ListGroup>
        <h2>My missions</h2>
        {reservedMissions.map((mission) => (
          <ListGroup.Item key={mission.mission_id}>{mission.mission_name}</ListGroup.Item>
        ))}
      </ListGroup>
    </Container>
  );
}
