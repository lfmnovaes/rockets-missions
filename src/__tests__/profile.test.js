import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import Profile from '../components/Profile';

describe('Group of Profile tests', () => {
  test('Test if Profile page is rendering correctly', () => {
    const missions = renderer
      .create(
        <Provider store={store}>
          <Profile />
        </Provider>,
      )
      .toJSON();
    expect(missions).toMatchSnapshot();
  });
});
