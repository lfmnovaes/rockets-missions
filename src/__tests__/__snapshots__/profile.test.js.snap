// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Group of Profile tests Test if Profile page is rendering correctly 1`] = `
<div
  className="d-flex justify-content-evenly container"
>
  <div
    className="list-group"
    onKeyDown={[Function]}
  >
    <h2>
      My Rockets
    </h2>
  </div>
  <div
    className="list-group"
    onKeyDown={[Function]}
  >
    <h2>
      My missions
    </h2>
  </div>
</div>
`;
